example/tests/test_serializers.py

```diff
@@ -193,3 +198,40 @@ def test_model_serializer_with_implicit_fields(self, comment, client):
 
         assert response.status_code == 200
         assert expected == response.json()
+
+
+class TestPolymorphicModelSerializer(TestCase):
+    def setUp(self):
+        self.project = ArtProjectFactory.create()
+
+    def test_polymorphic_model_serializer_passes_instance_to_child(self):
+        """
+        Ensure that `PolymorphicModelSerializer` is passing the instance to the
+        child serializer when initializing them
+        """
+        # Arrange
+        self.serializer_instance = None
+
+        def save(serializer_self):
+            """
+            Override `ArtProjectSerializer.save` to get the instance serializer,
+            which is later used assertion
+            """
+            self.serializer_instance = serializer_self.instance
+            return super(ArtProjectSerializer, serializer_self).save()
+
+        # Override `save` with our own method
+        ArtProjectSerializer.save = save
+
+        # Initialize a serializer that would partially update a model instance
+        data = {"artist": "Mark Bishop", "type": "artProjects"}
+        serializer = ProjectSerializer(
+            instance=self.project, data=data, partial=True
+        )
+        serializer.is_valid(raise_exception=True)
+
+        # Act
+        serializer.save()
+
+        # Assert
+        assert self.serializer_instance is not None
```

Comment:
I would move this into the save method and assert for self.project and not just for None.

Reply:
I added the asserts here and used `self.serializer_instance` because I wanted to have all the asserts in the test itself and not in inner methods, which I find easier to read and follow what the test is doing. But I'll move them to the `save` method, it's one of those decisions that are based purely in personal preference.